cub/util_device.cuh

```diff
@@ -402,6 +398,7 @@ CUB_RUNTIME_FUNCTION inline cudaError_t PtxVersionUncached(int& ptx_version)
 __host__ inline cudaError_t PtxVersionUncached(int& ptx_version, int device)
 {
     SwitchDevice sd(device);
+    (void)sd;
```

Comment:
SwitchDevice has the non-trivial constructor and destructor. Its object [won't cause unused variable warnings](https://godbolt.org/z/br344eE1b) in this context. Have you encountered an issue here?

Reply:
I can't remember if I actually hit an error here, but either way I still prefer to explicitly mark these as used. Unused variable warnings are notorious for false positives, so adding these defensively helps avoid situations where a new compiler release will create issues for our users.